src/phases/query/FilterWithComplexLogicalExpression.yml

```diff
@@ -0,0 +1,775 @@
+          OperationCommand:
+            Filter: {$expr: {$eq: ["$n1.a.b.c.d.z", 0]}}
+
+- Name: MatchExpressionMissingPathSuffix
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [16]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {"n1.a.b.c.d.z": 0}
+
+- ActorFromTemplate:
+    TemplateName: AggregationOrWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: AggregationExpressionFiftyClauseRootedOr
+      Repeat: 5
+      ActivePhase: 17
+      NumberOfClauses: 50
+
+- ActorFromTemplate:
+    TemplateName: MatchOrWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: MatchExpressionFiftyClauseRootedOr
+      Repeat: 5
+      ActivePhase: 18
+      NumberOfClauses: 50
+
+- ActorFromTemplate:
+    TemplateName: AggregationOrWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: AggregationExpressionOneHundredClauseRootedOr
+      Repeat: 1
+      ActivePhase: 19
+      NumberOfClauses: 100
+
+- ActorFromTemplate:
+    TemplateName: MatchOrWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: MatchExpressionOneHundredClauseRootedOr
+      Repeat: 1
+      ActivePhase: 20
+      NumberOfClauses: 100
+
+- ActorFromTemplate:
+    TemplateName: AggregationOrWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: AggregationExpressionTwoHundredClauseRootedOr
+      Repeat: 1
+      ActivePhase: 21
+      NumberOfClauses: 200
+
+- ActorFromTemplate:
+    TemplateName: MatchOrWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: MatchExpressionTwoHundredClauseRootedOr
+      Repeat: 1
+      ActivePhase: 22
+      NumberOfClauses: 200
+
+- ActorFromTemplate:
+    TemplateName: MatchAndWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: MatchExpressionFiftyClauseRootedAnd
+      Repeat: 5
+      ActivePhase: 23
+      NumberOfClauses: 50
+
+- ActorFromTemplate:
+    TemplateName: MatchAndWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: MatchExpressionOneHundredClauseRootedAnd
+      Repeat: 1
+      ActivePhase: 24
+      NumberOfClauses: 100
+
+- ActorFromTemplate:
+    TemplateName: MatchAndWithManyChildClausesTemplate
+    TemplateParameters:
+      Name: MatchExpressionTwoHundredClauseRootedAnd
+      Repeat: 1
+      ActivePhase: 25
+      NumberOfClauses: 200
+
+# Clear any pre-existing collection state.
+- Name: ClearCollectionAgain
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [26]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+# Populate collection for rooted $or index scan plans
+- Name: InsertDataForIndexScan
```

Comment:
I've no objection to extending the tests generally (this will be useful for M4), but just wanted to note that we won't be looking at any of the IndexScan metrics in Bonsai M2. We shouldn't need to do any additional work tagging-wise I think, since the metrics names are the same.